coredis/_utils.py

```diff
@@ -500,22 +503,48 @@ async def gather(
     awaitable5: Awaitable[T5],
     awaitable6: Awaitable[T6],
     /,
+    *,
+    return_exceptions: bool = False,
 ) -> tuple[T1, T2, T3, T4, T5, T6]: ...
 
 
 @overload
-async def gather(*awaitables: Awaitable[T1]) -> tuple[T1, ...]: ...
+async def gather(
+    *awaitables: Awaitable[T1],
+    return_exceptions: bool = False,
+) -> tuple[T1, ...]: ...
 
 
-async def gather(*awaitables: Awaitable[Any]) -> tuple[Any, ...]:
-    if len(awaitables) == 1:
```

Comment:
Can we keep this optimization? I think it's worth it to not have to create a task group in the first place

Reply:
Addressed in 12d2af7397a9bbe1243f14be036d65e3ec5ef760